src/sage/categories/finite_posets.py

```diff
@@ -1930,26 +1930,33 @@ def order_ideals_lattice(self, as_ideals=True, facade=None):
                 False
             """
             from sage.combinat.posets.lattices import LatticePoset
+            from sage.categories.finite_lattice_posets import FiniteLatticePosets
             if facade is None:
                 facade = self._is_facade
+
             if as_ideals:
                 from sage.misc.call import attrcall
                 from sage.sets.set import Set
                 ideals = [Set(self.order_ideal(antichain))
                           for antichain in self.antichains()]
-                return LatticePoset((ideals, attrcall("issubset")),
-                                    facade=facade)
-            else:
-                from sage.misc.cachefunc import cached_function
-                antichains = [tuple(a) for a in self.antichains()]
+                T = LatticePoset((ideals, attrcall("issubset")),
+                                 facade=facade)
+                T._refine_category_(FiniteLatticePosets().Distributive())
+                return T
+
+            from sage.misc.cachefunc import cached_function
+            antichains = [tuple(a) for a in self.antichains()]
+
+            @cached_function
+            def is_above(a, xb):
+                return any(self.is_lequal(xa, xb) for xa in a)
 
-                @cached_function
-                def is_above(a, xb):
-                    return any(self.is_lequal(xa, xb) for xa in a)
+            def compare(a, b):
+                return all(is_above(a, xb) for xb in b)
 
-                def compare(a, b):
-                    return all(is_above(a, xb) for xb in b)
-                return LatticePoset((antichains, compare), facade=facade)
+            T = LatticePoset((antichains, compare), facade=facade)
+            T._refine_category_(FiniteLatticePosets().Distributive())
```

Comment:
```suggestion
            cat = FiniteLatticePosets().Distributive()
            T = LatticePoset((antichains, compare), facade=facade, category=cat)
```
I think we should also briefly modify `LatticePoset` to accept a `category`. I think this is within the scope of this PR since before there was only one category it could belong to.

Reply:
Well, `LatticePoset` is already accepting a category. I have made so that this category is no longer ignored.

Reply:
Thank you. Yes, that's right (via a non-specific keyword arg).